cypress/integration/rich-text/RichTextEditor.spec.ts

```diff
@@ -658,6 +658,47 @@ describe('Rich Text Editor', { viewportHeight: 2000 }, () => {
       ].map((type) => getDropdownItem(type).get('button').should('not.be.disabled'));
     });
 
+    describe('Inserting Tables', () => {
+      it('replaces empty paragraphs with tables', () => {
+        insertTable();
+
+        richText.expectValue(
+          doc(
+            table(row(emptyHeader(), emptyHeader()), row(emptyCell(), emptyCell())),
+            emptyParagraph()
+          )
+        );
+      });
+
+      it('inserts new table below if paragraph is not empty', () => {
+        richText.editor.type('foo');
+
+        insertTable();
+
+        richText.expectValue(
+          doc(
+            paragraphWithText('foo'),
+            table(row(emptyHeader(), emptyHeader()), row(emptyCell(), emptyCell())),
+            emptyParagraph()
+          )
+        );
+      });
+
+      describe('Toolbar', () => {
+        const buttonsToDisableTable = ['quote', 'ul', 'ol'];
+
+        buttonsToDisableTable.forEach((button) => {
+          it(`should disable table button on toolbar if ${button} is selected`, () => {
```

Comment:
I think you meant
```suggestion
          it(`should disable ${button} button on toolbar if a table is focused`, () => {
```

Reply:
No, what I meant is right. That's what I wanted :) I don't want to add a table inside a blockquote or a list, so the table button should be disabled in those cases.

Reply:
Sorry, I actually got this test a bit wrong. With that in mind, my suggestion above also didn't really make sense. The `${button}` variable naming also threw me a bit off track here as I was thinking "button" while we're referring to the actual element being selected, resulting in the table button being disabled.